src/zenml/client.py

```diff
@@ -2083,7 +2094,7 @@ def update_stack_component(
 
             validated_config = validate_stack_component_config(
```

Comment:
I realize this function is used in different contexts, but for instance, here, we already have a component model that has a flavor model in its resource but we pass the flavor name and end up querying for the flavor again.

Reply:
Updated the function to allow a flavor response in cases where we already have it

Reply:
Nice 👍